earthpy/plot.py

```diff
@@ -450,6 +450,11 @@ def hist(
     """
 
     # If the array is 3 dimensional setup grid plotting
+    if np.ma.is_masked(arr):
+        arrlis = []
+        for i in range(arr.ndim):
```

Comment:
@nkorinek do you need to loop through the data like this? can't you just call `arr[~arr.mask]` ?

Reply:
I don't just call `arr[~arr.mask]` bc that always returns a one dimensional array, even if a multi-dimensional array is passed into the function. For example, an array of shape `(7, 1941, 2158)` will have a shape of `(28901425,)` after that function is run on it.

Reply:
ok gotcha!! thank you for the clarification @nkorinek !!